Bottom: f5629b68520a9714d3917344c29b8f521738a7f2
Top:    a1173dcb4cc4533a932dc2d424c6309277dcd37b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-05 02:15:25 +0100

hid/lesstif: Process all pending events in lesstif_process

I just found out about the XtAppPending() function.


---

diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index a4b9975..08e2ad9 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -3922,11 +3922,11 @@ lesstif_progress (int so_far, int total, const char *message)
   if (!started)
     XtManageChild (progress_dialog);
 
-  /* Dispatch one event - ideally we would keep dispatching until we
-   * were about to block, but I can't see how to do this with Xt
-   */
-  XtAppNextEvent (app_context, &e);
-  XtDispatchEvent (&e);
+  while (XtAppPending (app_context))
+    {
+      XtAppNextEvent (app_context, &e);
+      XtDispatchEvent (&e);
+    }
 
   /* Note the time we did this */
   gettimeofday (&last_time, NULL);
